src/main/java/seedu/address/model/event/StartTime.java

```diff
@@ -3,55 +3,55 @@
 import static java.util.Objects.requireNonNull;
 import static seedu.address.commons.util.AppUtil.checkArgument;
 
+import java.text.DecimalFormat;
+import java.time.DateTimeException;
 import java.time.LocalTime;
-import java.time.format.DateTimeFormatter;
-import java.time.format.DateTimeParseException;
+
 
 /**
  * Represents an Event's starting time in the address book.
  * Guarantees: immutable; is valid as declared in {@link #isValidStartTime(String)}
  */
 public class StartTime implements Comparable<StartTime> {
 
-    public static final String MESSAGE_CONSTRAINTS = "Start time must be in format: hh/mm/AM, hh/mm/PM or hh:mm";
-
-    //for checking if valid input date format
-    private static final DateTimeFormatter checkFormatter = DateTimeFormatter
-            .ofPattern("[HH:MM][hh/mm/a][h/mm/a][hh/m/a][h/m/a]");
-
-    //for changing to storage friendly format
-    private static final DateTimeFormatter logFormatter = DateTimeFormatter.ofPattern("hh/mm/a");
-
-    //for changing to user-readable format
-    private static final DateTimeFormatter outputFormatter = DateTimeFormatter.ofPattern("hh:mm a");
-
+    public static final String MESSAGE_FORMAT_CONSTRAINTS = "Start time must be in format: hh:mm.";
+    public static final String MESSAGE_VALUE_CONSTRAINTS = "%s exceeds the range of valid time value.";
+    private static final DecimalFormat df = new DecimalFormat("00");
     public final LocalTime time;
 
 
     /**
      * Constructs a {@code StartTime}.
-     *
      * @param startTime A valid start time.
      */
     public StartTime(String startTime) {
         requireNonNull(startTime);
         assert !startTime.isBlank();
-        checkArgument(isValidStartTime(startTime), MESSAGE_CONSTRAINTS);
-        this.time = LocalTime.parse(startTime, checkFormatter);
+        checkArgument(isValidStartTime(startTime) != 0, MESSAGE_FORMAT_CONSTRAINTS);
+        checkArgument(isValidStartTime(startTime) != -1, String.format(MESSAGE_VALUE_CONSTRAINTS, startTime));
+        String [] parsedTime = startTime.split(":", 2);
+        this.time = LocalTime.of(Integer.parseInt(parsedTime[0]), Integer.parseInt(parsedTime[1]), 0);
     }
 
     /**
-     * Returns true if a given string is a valid StartTime input.
-     * @return boolean
+     * Checks if a given string is a valid StartTime input.
+     * @return int value, 0 represent invalid time format, -1 represents invalid time value, 1 represents valid time.
      */
-    //found from https://mkyong.com/java/how-to-check-if-date-is-valid-in-java/
-    public static boolean isValidStartTime(String test) {
+    public static int isValidStartTime(String test) {
```

Comment:
Hmm I'm not sure if it's a good idea to convert this method to return an integer instead of a boolean. I understand you're doing this to distinguish between whether the time format is invalid or whether the time value is invalid. But doing so makes this method inconsistent with the other `isValidXXX()` methods in other classes since they all return a boolean.

Having said that, I can't think of any good alternatives to solve this issue. 😅 So I'm fine if you don't change this.

The only thing I can suggest is to consider using regex to validate the input string instead of actually trying to create the `LocalTime` object with the given inputs. While this doesn't solve the original problem, it could help shorten the method and make it more readable. But I'm not sure if you're comfortable with the regex syntax, so it's fine if you don't do so.

Reply:
I also thought about the inconsistency issue when changing the return type of this method. Maybe I can try adding another method in StartTime class named `isValidStartTimeValue(String test)` to check if the value is valid, so that`isValidStartTime(String test)` only needs to check the time format and it can preserve the boolean return value.